.github/workflows/docs.yml

```diff
@@ -44,4 +45,58 @@ jobs:
         sed -i -e 's/-j auto/-j 1/' Makefile
         make html
         
-        mv build/html/* "${RUNNER_DOCS_DIR}"
+        cp -r build/html "${RUNNER_ARTIFACT_DIR}"
+        
+        # On PRs we also want to upload the docs into our S3 bucket for preview.
+        if [[ ${{ github.event_name == 'pull_request' }} ]]; then
+          cp -r build/html/* "${RUNNER_DOCS_DIR}"
+        fi
+
+  upload:
+    # FIXME: this also needs to happen on created releases
+#    # FIXME: enable this before the PR is merged
+#    if: github.event_name == 'push' && github.event.push.base_ref == 'main'
+    needs: build
+    uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    with:
+      repository: pytorch/vision
+      download-artifact: docs
+      ref: gh-pages
+      script: |
+        set -euo pipefail
+        
+        TARGET="main"
+
+        mkdir -p "${TARGET}"
+        rm -rf "${TARGET}"/*
+        mv "${RUNNER_ARTIFACT_DIR}"/html/* "${TARGET}"
+        git add "${TARGET}" || true
+        
+        if [[ "${TARGET}" == main ]]; then
+          mkdir -p _static
+          rm -rf _static/*
+          cp -r "${TARGET}"/_static/* _static
+          git add _static || true
+        fi
+        
+        git config user.email "soumith+bot@pytorch.org"
+        git config user.name "pytorchbot"
+        git commit -m "auto-generating sphinx docs" || true
+
+        # FIXME: this needs to come from ${{ secrets }}
+        export GITHUB_PYTORCHBOT_TOKEN=FOO
+        {
+          echo "machine github.com"
+          echo "login pytorchbot"
+          echo "password ${GITHUB_PYTORCHBOT_TOKEN}"
+        } > ~/.netrc
```

Comment:
Previously, we had `GITHUB_PYTORCHBOT_TOKEN` available as environment variable. I already checked and it is not available through the GH secrets. On CIrcleCI we had

https://github.com/pytorch/vision/blob/82cf540b45fae6bc83cfff10db9041c3a1aeec5d/.circleci/config.yml#L919-L927

Not sure how to access the org-member context.

Reply:
Maybe not, I honestly have not checked. Maybe we don't need it.